src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -127,6 +137,8 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
         // `lib` in the compiler and we're not sure which we'll see.
         crate_types.insert("bin".to_string());
         crate_types.insert("rlib".to_string());
+        // TODO comment
+        crate_types.insert("metadata".to_string());
```

Comment:
The failures on stable (via Travis) point to we'll need to gracefully handle the situation where the compiler we're running against doesn't support `--crate-type metadata`.